### PR TITLE
Use x-ms-date for HMAC authentication

### PR DESCRIPTION
**Pull request by JianpingChen**
Use "x-ms-date" for HMAC signed headers instead of "date" header.
Add a new header, "x-ms-hmac-string-to-sign-base64", for debugging purpuse.

### CODE REVIEW

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/implementation/HmacAuthenticationPolicy.java [hunk 1]
```diff
@@ -37,11 +37,12 @@
  * HttpPipelinePolicy to append CommunicationClient required headers
  */
 public final class HmacAuthenticationPolicy implements HttpPipelinePolicy {
-    private static final String DATE_HEADER = "date";
+    private static final String X_MS_DATE_HEADER = "x-ms-date";
+    private static final String X_MS_STRING_TO_SIGH_HEADER = "x-ms-hmac-string-to-sign-base64";
```
Comment (DominikMe) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/implementation/HmacAuthenticationPolicy.java:41]: typo :-)
```suggestion
    private static final String X_MS_STRING_TO_SIGN_HEADER = "x-ms-hmac-string-to-sign-base64";
```

Comment (JianpingChen) [reply sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/implementation/HmacAuthenticationPolicy.java]: Fixed.

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/implementation/HmacAuthenticationPolicy.java [hunk 2]
```diff
@@ -38,10 +38,12 @@
  */
 public final class HmacAuthenticationPolicy implements HttpPipelinePolicy {
     private static final String DATE_HEADER = "date";
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/implementation/HmacAuthenticationPolicy.java:40]: Should you remove this now?

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/implementation/HmacAuthenticationPolicy.java [hunk 3]
```diff
@@ -127,13 +129,14 @@ public HmacAuthenticationPolicy(AzureKeyCredential clientCredential) {
         headers.put(CONTENT_HASH_HEADER, contentHash);
         String utcNow = OffsetDateTime.now(ZoneOffset.UTC)
             .format(HMAC_DATETIMEFORMATTER_PATTERN);
+        headers.put(X_MS_DATE_HEADER, utcNow);
         headers.put(DATE_HEADER, utcNow);
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/implementation/HmacAuthenticationPolicy.java:133]: Is this still needed?

**Review by ankitarorabit — APPROVED** [review]

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/implementation/HmacAuthenticationPolicy.java (hunk 3 above)
Comment (JianpingChen) [reply sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/implementation/HmacAuthenticationPolicy.java:133]: Yes. We want to have log entries on service side to see if LinkedIn proxies actually modified Date header.

**Pull request merged**